### PR TITLE
IOS-XR: Add iosxr_acls model

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

### CODE REVIEW

File: models/iosxr/acls/iosxr_acls.yaml [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+                          type: bool
+                        traceroute:
+                          description: Traceroute
+                          type: bool
+                        ttl_exceeded:
+                          description: TTL exceeded
+                          type: bool
+                        unreachable:
+                          description: All unreachables
+                          type: bool
```
Comment (GomathiselviS) [inline models/iosxr/acls/iosxr_acls.yaml]: Do we have icmpv6 and other v6 protocols for ipv6 access-list ? If yes and if they have sub options, they should be added under protocol_options

File: models/iosxr/acls/iosxr_acls.yaml [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+                      type: str
+                    wildcard_bits:
+                      description:
+                        - The Wildcard bits to apply to source address.
+                      type: str
+                    any:
+                      description:
+                        - Match any source address.
+                      type: bool
+                    port_protocol:
```
Comment (justjais) [inline models/iosxr/acls/iosxr_acls.yaml]: this should be `operators` instead of `port_protocol`

File: models/iosxr/acls/iosxr_acls.yaml [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+                        start:
+                          description: Start of the packet length range
+                          type: int
+                        end:
+                          description: End of the packet length range
+                          type: int
+                precedence:
+                  description: Match packets with given precedence value
+                  type: str
+                ttl:
```
Comment (justjais) [inline models/iosxr/acls/iosxr_acls.yaml]: dscp, fragments, ttl and tos should be the list of diff param or should be separate param

Comment (NilashishC) [reply models/iosxr/acls/iosxr_acls.yaml]: You mean should these be top level keys or under another key called `match`?

File: models/iosxr/acls/iosxr_acls.yaml (hunk 1 above)
Comment (NilashishC) [reply models/iosxr/acls/iosxr_acls.yaml]: Done. Argspec updated in https://github.com/ansible/ansible/pull/66207/files.

**Pull request merged**